### PR TITLE
IoT Hub - X509CA support 

### PR DESCRIPTION
**Pull request by mastermanu**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by alvadb — CHANGES_REQUESTED** [review]
There are multiple linter errors (https://travis-ci.org/Azure/azure-rest-api-specs/jobs/278476257). Please fix.

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json [hunk 1]
```diff
@@ -1013,9 +1013,492 @@
...
+        "operationId": "Certificates_GenerateVerificationCode",
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
```
Comment (alvadb) [inline specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: Please move this to the global parameter section and reference throughout.

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json [hunk 2]
```diff
@@ -1013,9 +1013,492 @@
...
+            "description": "The name of the resource group that contains the IoT hub.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resourceName",
```
Comment (alvadb) [inline specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: Please move this to the global parameter section and reference throughout.

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json [hunk 3]
```diff
@@ -1013,9 +1013,492 @@
...
+            "in": "path",
+            "description": "The name of the IoT hub.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "certificateName",
```
Comment (alvadb) [inline specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: Please move this to the global parameter section and reference throughout.

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json (hunk 2 above)
Comment (mastermanu) [reply specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: Fixed

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json (hunk 1 above)
Comment (mastermanu) [reply specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: Fixed

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json (hunk 3 above)
Comment (mastermanu) [reply specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: Fixed

**Comment by mastermanu** [comment]
@alvadb - Please see: https://github.com/Azure/azure-rest-api-specs/pull/1514. (This was the previous PR, but the employee has left the company so I created a new PR). It was agreed that the Linter warnings do not apply in this particular instance and was signed off by Anu

**Comment by ravbhatnagar** [comment]
@mastermanu @manusr I don't see the changes that were requested in the older PR. Properties like subjectname etc should go inside the properties bag. Please make sure the certificate resource complies with the basic shape needed for an ARM resource. Since this is likely a proxy resource, it will have id, name and type properties and all other peoperties will go inside the properties bag

**Review by alvadb — APPROVED** [review]

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json [hunk 4]
```diff
@@ -2138,6 +2441,28 @@
       "description": "The version of the API.",
       "required": true,
       "type": "string"
+    },
+	"resourceGroupName": {
```
Comment (amarzavery) [inline specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: https://github.com/Azure/azure-openapi-validator/issues/84

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json [hunk 5]
```diff
@@ -1013,9 +881,359 @@
         },
         "deprecated": false
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/certificates": {
+      "get": {
+        "tags": [ "Certificates" ],
+        "summary": "Get the certificate list.",
+        "description": "Returns the list of certificates.",
+        "operationId": "Certificates_ListByIoTHub",
```
Comment (andrew-buckley) [inline specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: should be Certificates_ListByIotHub, autorest generated sdks are getting confused.

File: specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json (hunk 4 above)
Comment (andrew-buckley) [reply specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: @amarzavery I've added "x-ms-parameter-location": "method" to the objects that describe parameters that are only used in methods, however I'm losing the "credentials" parameter `IotHubClientConfiguration(credentials, subscription_id, base_url)` has now become `IotHubClientConfiguration(subscription_id, base_url)` and this seems to be causing several issues. Any thoughts on how to fix this?

Comment (andrew-buckley) [reply specification/iothub/resource-manager/Microsoft.Devices/2017-07-01/iothub.json]: Ah seems to be a autorest option that might be new

**Pull request merged**